src/index.js

```diff
@@ -1,10 +1,223 @@
+  } else if (!isValidRootInsert(refNodeInsert)) {
+    err = 'refNodeInsert is missing or invalid';
+  }
+
+  if (err) {
+    throw new Error(err);
+  }
+};
+
+/**
+ * Initializes a player and returns it.
+ *
+ * @private
+ * @param  {Object} params
+ *         A parameters object. See README for details.
+ *
+ * @param  {Element} embed
+ *         An element that will be passed to the `bc()` function.
+ *
+ * @return {Object}
+ *         A Video.js player.
+ */
+const initPlayer = (params, embed) => {
+  const {playerId, embedId} = params;
+  const bc = window.bc[`${playerId}_${embedId}`] || window.bc;
+
+  if (!bc) {
+    return new Error(`missing bc function for ${playerId}`);
+  }
+
+  return bc(embed, params.options);
+};
+
+/**
+ * Loads a player from CDN and embeds it.
+ *
+ * @private
+ * @param  {Object} params
+ *         A parameters object. See README for details.
+ *
+ * @param  {Function} resolve
+ *         A function to call if a player is successfully initialized.
+ *
+ * @param  {Function} reject
+ *         A function to call if a player fails to be initialized.
+ */
+const loadPlayer = (params, resolve, reject) => {
+  checkParams(params);
+
+  const embed = createEmbed(params);
+
+  // If this is an iframe, all we need to do is create the embed code and
+  // inject it. Because there is no reliable way to hook into an iframe from
+  // the parent page, we simply resolve immediately upon creating the embed.
+  if (params.embedType === EMBED_TYPE_IFRAME) {
+    resolve(embed);
+    return;
+  }
+
+  const src = getUrl(params);
+
+  // If we've already downloaded this script, we should have the proper `bc`
+  // global and can bypass the script creation process.
+  if (downloadedScripts[src]) {
+    resolve(initPlayer(params, embed));
+    return;
+  }
+
+  const script = document.createElement('script');
+
+  script.onload = () => {
+    downloadedScripts[src] = 1;
+    resolve(initPlayer(params, embed));
+  };
+
+  script.onerror = () => {
+    reject(new Error('player script could not be downloaded'));
+  };
+
+  script.async = true;
+  script.charset = 'utf-8';
+  script.src = src;
+
+  document.head.appendChild(script);
+};
 
 /**
  * A function for asynchronously loading a Brightcove Player into a web page.
+ *
+ * @param  {Object} parameters
+ *         A parameters object. See README for details.
+ *
+ * @return {Promise|undefined}
+ *         A Promise, if possible.
  */
-function brightcovePlayerLoader() {}
+const brightcovePlayerLoader = (parameters) => {
+  const params = Object.assign({}, DEFAULTS, parameters);
+
+  if (typeof params.Promise === 'function') {
+    return new params.Promise((resolve, reject) => {
+      loadPlayer(params, resolve, reject);
+    });
+  }
+
+  loadPlayer(params, () => {}, (err) => {
+    throw err;
+  });
+};
 
-brightcovePlayerLoader.VERSION = VERSION;
+// Define some read-only constants on the exported function.
+[
+  ['EMBED_TYPE_IN_PAGE', EMBED_TYPE_IN_PAGE],
```

Comment:
Should this just be an object? should we just export all constants?

Reply:
I like them being properties of library itself rather than having an intermediate object (i.e. I like `brightcovePlayerLoader.EMBED_TYPE_IN_PAGE` rather than `brightcovePlayerLoader.constants.EMBED_TYPE_IN_PAGE`).

Reply:
Turns out you can't do that. If you have named exports, Rollup will make them writable properties of the default export, which isn't what I wanted. I wanted non-writable/configurable properties.